omas/omas_physics.py

```diff
@@ -437,54 +445,48 @@ def summary_total_powers(ods, update=True):
-    power_dict = {'total': [], 'nbi': [], 'ec': [], 'lh': [], 'ic': []}
-    q_init = numpy.zeros(len(sources['0.profiles_1d.0.grid.volume']))
     q_dict = {
-        'total': copy.deepcopy(q_init),
+        'total_heating': copy.deepcopy(q_init),
         'nbi': copy.deepcopy(q_init),
         'ec': copy.deepcopy(q_init),
         'lh': copy.deepcopy(q_init),
         'ic': copy.deepcopy(q_init),
+        'fusion': copy.deepcopy(q_init)
     }
-    ignore_indices = list(range(100, 108)) + list(range(900, 910))
 
-    for time_index in sources['0.profiles_1d']:
-        vol = sources[f'0.profiles_1d.{time_index}.grid.volume']
+    for time_index in sources[0]['profiles_1d']:
+        vol = sources[0]['profiles_1d'][time_index]['grid']['volume']
         for source in sources:
-            if sources[f'{source}.identifier.index'] in ignore_indices:
-                # Skip the combined sources to prevent double counting
-                continue
-            if f'{source}.profiles_1d.{time_index}.electrons.energy' in sources:
-                q_dict['total'] += sources[f'{source}.profiles_1d.{time_index}.electrons.energy']
-                if sources[f'{source}.identifier.index'] in index_dict:
-                    q_dict[index_dict[sources[f'{source}.identifier.index']]] += sources[
-                        f'{source}.profiles_1d.{time_index}.electrons.energy'
-                    ]
-            if f'{source}.profiles_1d.{time_index}.total_ion_energy' in sources:
-                q_dict['total'] += sources[f'{source}.profiles_1d.{time_index}.total_ion_energy']
-                if sources[f'{source}.identifier.index'] in index_dict:
-                    q_dict[index_dict[sources[f'{source}.identifier.index']]] += sources[
-                        f'{source}.profiles_1d.{time_index}.total_ion_energy'
-                    ]
-
-        for key, value in power_dict.items():
-            power_dict[key].append(numpy.trapz(q_dict[key], x=vol))
+            source_1d = sources[source]['profiles_1d'][time_index]
+            if sources[source]['identifier.index'] in index_dict:
+                if 'electrons' in source_1d and 'energy' in source_1d['electrons']:
+                    q_dict['total_heating'] += source_1d['electrons']['energy']
+                    if sources[source]['identifier.index'] in power_dict:
+                        q_dict[index_dict[sources[source]['identifier.index']]] += source_1d['electrons']['energy']
+                if 'total_ion_energy' in source_1d:
+                    q_dict['total_heating'] += source_1d['total_ion_energy']
+                    if sources[source]['identifier.index'] in power_dict:
+                        q_dict[index_dict[sources[source]['identifier.index']]] += source_1d['total_ion_energy']
 
     for key, value in power_dict.items():
+        power_dict[key].append(numpy.trapz(q_dict[key], x=vol))
         if numpy.sum(value) > 0:
-            if key == 'total':
+            if key is 'total_heating':
```

Comment:
cannot compare strings with `is`. This will always evaluate to false.

Reply:
yeah you are right 
```if 'total_heating' in key```
 is way better